pkg/services/guardian/provider.go

```diff
@@ -14,9 +14,21 @@ type Provider struct{}
 func ProvideService(store *sqlstore.SQLStore, ac accesscontrol.AccessControl, permissionsServices accesscontrol.PermissionsServices, features featuremgmt.FeatureToggles) *Provider {
 	if features.IsEnabled(featuremgmt.FlagAccesscontrol) {
 		// TODO: Fix this hack, see https://github.com/grafana/grafana-enterprise/issues/2935
-		New = func(ctx context.Context, dashId int64, orgId int64, user *models.SignedInUser) DashboardGuardian {
-			return NewAccessControlDashboardGuardian(ctx, dashId, user, store, ac, permissionsServices)
-		}
+		InitACLGuardianWithStore(store, ac, permissionsServices)
+	} else {
+		InitGuardianWithStore(store)
 	}
 	return &Provider{}
 }
+
+func InitGuardianWithStore(store sqlstore.Store) {
```

Comment:
Is this what we discussed yesterday? For internal calls, we have the implementation without permission check. For API calls, we have the implementation with permission check. Then what the feature flag FlagAccesscontrol here for?

Reply:
No, this switches between using the service-specific `dashboard_acl` table and a fine-grained access control based compatibility layer for the guardian 🙂